Source/src/ui/WindowHierarchy.cpp

```diff
@@ -28,27 +28,43 @@ void Hachiko::WindowHierarchy::CleanUp()
 
 void Hachiko::WindowHierarchy::DrawHierarchyTree(const GameObject* game_object)
 {
-    DrawChildren(game_object);
+    for (const auto go : game_object->children)
+    {
+       RecursiveDraw(go);
+    }
+
     if (!App->input->GetMouseButton(SDL_BUTTON_LEFT))
     {
         dragged_object = nullptr;
     }
 }
 
-void Hachiko::WindowHierarchy::DrawChildren(const GameObject* game_object)
+bool Hachiko::WindowHierarchy::RecursiveDraw(GameObject* game_object)
 {
-    for (const auto go : game_object->children)
+    bool stop = false;
+
+    const char* str = strstr(game_object->name.c_str(), "$AssimpFbx$");
+    if (str != nullptr)
+    {
+        for (GameObject* go : game_object->children)
+        {
+            if ((stop = RecursiveDraw(go)) == true)
+            {
+                break;
+            }
+        }
+    }
+    else
     {
-        ImGui::PushID(go);
-        DrawGameObject(go);
-        ImGui::PopID();
+        stop = DrawGameObject(game_object, stop);
     }
+
+    return stop;
 }
 
-// TODO: Refactor to simplify function
-void Hachiko::WindowHierarchy::DrawGameObject(GameObject* game_object)
+bool Hachiko::WindowHierarchy::DrawGameObject(GameObject* game_object, bool stop)
 {
-    ImGuiTreeNodeFlags flags = ImGuiTreeNodeFlags_OpenOnArrow | ImGuiTreeNodeFlags_DefaultOpen;
+    ImGuiTreeNodeFlags flags = ImGuiTreeNodeFlags_OpenOnArrow;
```

Comment:
It's actually way better if the GameObject nodes are expanded by default.
```suggestion
    ImGuiTreeNodeFlags flags = ImGuiTreeNodeFlags_OpenOnArrow | ImGuiTreeNodeFlags_DefaultOpen;
```

Reply:
Ou you are sure? there are a loooooooooooooooot of bones, only with zombuny you need to do a lot of scroll it's a pain

Reply:
Okayyy